src/scripts/DsDisclosure.js

```diff
@@ -7,12 +7,15 @@ module.exports = class extends DsElement {
     }
 
     static get observedAttributes () {
-        return [ 'aria-expanded' ];
+        return [ 
+            'aria-expanded',
+            'disabled',
+        ];
     }
 
     connectedCallback () {
         this.$setAttribute('role', 'button');
-        this.$setAttribute('tabindex', 0);
+        this.setAttribute('tabindex', 0);
```

Comment:
Removing check on user-set value, as it conflicts with `[disabled]` functionality.